crates/partition-store/src/invocation_status_table/mod.rs

```diff
@@ -135,15 +135,17 @@ pub(crate) async fn run_invocation_status_v1_migration(storage: &mut PartitionSt
         )
         .map_err(|_| StorageError::OperationalError)?;
 
+    let mut tx = storage.transaction();
     while let Some(res) = iterator.next().await {
         let (key, value) = res?;
         put_invocation_status(
-            storage,
+            &mut tx,
             &InvocationId::from_parts(*key.partition_key_ok_or()?, *key.invocation_uuid_ok_or()?),
             &value.0,
         )?;
-        storage.delete_key(&key)?;
+        tx.delete_key(&key)?;
     }
+    tx.commit().await?;
```

Comment:
We definitely want to commit periodically, say every `MIGRATION_BATCH_SIZE` = a few thousand records. There is a potentially very large number of records to migrate and we might blow up if we try stuff them all into a single write batch.